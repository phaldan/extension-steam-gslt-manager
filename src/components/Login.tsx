import * as React from 'react';
import { Modal } from 'react-bootstrap';

interface LoginProps {
  show: boolean;
}

export default class Login extends React.Component<LoginProps> {
  public render() {
    const { show } = this.props;
    return (
      <span>
        <Modal show={show}>
          <Modal.Header>
            <Modal.Title>
              Login
            </Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <div className="panel panel-info" style={{margin: 0}}>
                <div className="panel-heading">
                  <div className="page-header text-center">
                    <h4>
                      Please <a href="https://steamcommunity.com/login/home/">login</a>
                      &nbsp; into your Steam account for using this browser extension!
                    </h4>
                  </div>
                </div>
            </div>
          </Modal.Body>
        </Modal>
      </span>
    );
  }
}
