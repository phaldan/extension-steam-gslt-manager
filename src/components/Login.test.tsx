import { shallow } from 'enzyme';
import React from 'react';
import Login from './Login';

function createComponent(show) {
    return shallow(<Login show={show} />);
}

describe('<Login>', () => {
    test('show modal', () =>  {
        const target = createComponent(true);
        expect(target).toMatchSnapshot();
    });

    test('hidden modal', () =>  {
        const target = createComponent(false);
        expect(target).toMatchSnapshot();
    });
});
