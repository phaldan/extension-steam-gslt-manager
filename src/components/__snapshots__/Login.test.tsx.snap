// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> hidden modal 1`] = `
<span>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Login
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        className="panel panel-info"
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <div
          className="panel-heading"
        >
          <div
            className="page-header text-center"
          >
            <h4>
              Please 
              <a
                href="https://steamcommunity.com/login/home/"
              >
                login
              </a>
                into your Steam account for using this browser extension!
            </h4>
          </div>
        </div>
      </div>
    </ModalBody>
  </Modal>
</span>
`;

exports[`<Login> show modal 1`] = `
<span>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Login
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <div
        className="panel panel-info"
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <div
          className="panel-heading"
        >
          <div
            className="page-header text-center"
          >
            <h4>
              Please 
              <a
                href="https://steamcommunity.com/login/home/"
              >
                login
              </a>
                into your Steam account for using this browser extension!
            </h4>
          </div>
        </div>
      </div>
    </ModalBody>
  </Modal>
</span>
`;
